hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -1185,85 +1194,72 @@ public void pruneOlderSnapshotsWithCompactionHistory() {
 
     try (BootstrapStateHandler.Lock lock = getBootstrapStateLock().lock()) {
       removeSstFiles(sstFileNodesRemoved);
-      deleteOlderSnapshotsCompactionFiles(olderSnapshotsLogFilePaths);
+      removeKeyFromCompactionLogTable(keysToRemove);
     } catch (InterruptedException e) {
       throw new RuntimeException(e);
     }
   }
 
-  /**
-   * Deletes the SST files from the backup directory if exists.
-   */
-  private void removeSstFiles(Set<String> sstFileNodes) {
-    for (String sstFileNode: sstFileNodes) {
-      File file =
-          new File(sstBackupDir + "/" + sstFileNode + SST_FILE_EXTENSION);
-      try {
-        Files.deleteIfExists(file.toPath());
-      } catch (IOException exception) {
-        LOG.warn("Failed to delete SST file: " + sstFileNode, exception);
-      }
-    }
-  }
 
   /**
-   * Returns the list of compaction log files which are older than allowed
-   * max time in the compaction DAG.
+   * Returns the list of input files from the compaction entries which are
+   * older than the maximum allowed in the compaction DAG.
    */
-  private List<Path> getOlderSnapshotsCompactionLogFilePaths() {
+  private synchronized Pair<List<String>, List<byte[]>> getOlderFileNodes() {
```

Comment:
```suggestion
  private synchronized Pair<Set<String>, List<byte[]>> getOlderFileNodes() {
```
Return Set of string instead of list

Reply:
Changed it to set as suggested.